### DISCUSSION TITLE
Why are diagnostics not working for ``.md`` files?

### DISCUSSION THREAD

**Discussion by robg-test** [LazyVim/LazyVim#2384] [Q&A] [answered]
There are no diagnostics (virtual text) being written out and no errors detected in a markdown file I have created.

![image](https://github.com/LazyVim/LazyVim/assets/106234256/96082365-17b8-45c5-8a74-dae8cf692a1d)

The markdown server is definitely running with the markdown server attached as per below
 
```
 Language client log: /home/robert/.local/state/nvim/lsp.log
 Detected filetype:   markdown
 
 1 client(s) attached to this buffer: 
 
 Client: marksman (id: 1, bufnr: [3])
 	filetypes:       markdown, markdown.mdx
 	autostart:       true
 	root directory:  Running in single file mode.
 	cmd:             /home/robert/.local/share/nvim/mason/bin/marksman server
 
 Configured servers list: gopls, omnisharp, marksman, pyright, docker_compose_language_service, terraformls, yamlls, eslint, lua_ls, jsonls, tsserver, jdtls, dockerl

```

The ``LspLog`` is indicating an error but I presume this is just highlighting an issue with my document and not an error with writing out the diagnostics?

```
[START][2024-01-19 20:46:40] LSP logging initiated
[ERROR][2024-01-19 20:46:40] .../vim/lsp/rpc.lua:677	"rpc"	"/home/robert/.local/share/nvim/mason/bin/marksman"	"stderr"	"[20:46:40 INF] <LSP Entry> Starting Marksman LSP server: {}\n"
[START][2024-01-19 20:46:49] LSP logging initiated
[ERROR][2024-01-19 20:46:49] .../vim/lsp/rpc.lua:677	"rpc"	"/home/robert/.local/share/nvim/mason/bin/marksman"	"stderr"	"[20:46:49 INF] <LSP Entry> Starting Marksman LSP server: {}\n"
```

Diagnostics are toggled on using``<leader>ud`` 

I had this working on my Windows LazyVim setup however it doesn't seem to work on my Ubuntu 22.04 Laptop? Is there some toggle or configuration I'm missing?

**Comment by dpetka2001** [comment]
![2024-01-19_23-11](https://github.com/LazyVim/LazyVim/assets/12776461/f5cc6684-cf23-4c54-bb20-7b9950285130)
This is what I get with your example. Something must be wrong with your configuration, since you got it working on a different environment.

**Comment by robg-test** [comment, marked as answer]
So got to the bottom of this. As it turns out Markdown-lint the configured LSP that provides these details is not compatible with Node v12.
I installed node via ``apt-get install nodejs`` which on Ubuntu installs Node 12, doh. 

There are some warnings of incompatibility which you can see when trying to globally install the LSP's via NPM, so you need to have NodeJS v18 or higher in order for this to work.

There's no LSP logging for MarkdownLint nor any indication anywhere that the linting is non-functioning but then again Node 12 is slightly out of support :smile:
